webhooks/pod_webhook_test.go

```diff
@@ -1,657 +1,533 @@
+			want: []string{"false"},
+		}, {
+			name: "annotated with enabled index",
+			o:    &corev1.Pod{ObjectMeta: metav1.ObjectMeta{Annotations: map[string]string{"openfeature.dev/allowkubernetessync": "true"}}},
+			want: []string{"true"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := OpenFeatureEnabledAnnotationIndex(tt.o); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("OpenFeatureEnabledAnnotationIndex() = %v, want %v", got, tt.want)
+			}
+		})
 	}
-	err = k8sClient.Create(testCtx, clusterRoleBinding)
-	Expect(err).ShouldNot(HaveOccurred())
-
-	existingPod := testPod(existingPod1Name, existingPod1ServiceAccountName, map[string]string{
-		fmt.Sprintf("%s/%s", OpenFeatureAnnotationPrefix, EnabledAnnotation):                  "true",
-		fmt.Sprintf("%s/%s", OpenFeatureAnnotationPrefix, FeatureFlagConfigurationAnnotation): fmt.Sprintf("%s/%s", mutatePodNamespace, featureFlagConfigurationName),
-		fmt.Sprintf("%s/%s", OpenFeatureAnnotationPrefix, AllowKubernetesSyncAnnotation):      "true",
-	})
-	err = k8sClient.Create(testCtx, existingPod)
-	Expect(err).ShouldNot(HaveOccurred())
-
-	existingPod = testPod(existingPod2Name, existingPod2ServiceAccountName, map[string]string{
-		OpenFeatureAnnotationPrefix: "enabled",
-		fmt.Sprintf("%s/%s", OpenFeatureAnnotationPrefix, FeatureFlagConfigurationAnnotation): fmt.Sprintf("%s/%s", mutatePodNamespace, featureFlagConfigurationName),
-		fmt.Sprintf("%s/%s", OpenFeatureAnnotationPrefix, AllowKubernetesSyncAnnotation):      "true",
-	})
-	err = k8sClient.Create(testCtx, existingPod)
-	Expect(err).ShouldNot(HaveOccurred())
 }
 
-func setupMutatePodResources() {
-	svcAccount := &corev1.ServiceAccount{}
-	svcAccount.Namespace = mutatePodNamespace
-	svcAccount.Name = defaultPodServiceAccountName
-	err := k8sClient.Create(testCtx, svcAccount)
-	Expect(err).ShouldNot(HaveOccurred())
-
-	ffConfig := &v1alpha1.FeatureFlagConfiguration{}
-	ffConfig.Namespace = mutatePodNamespace
-	ffConfig.Name = featureFlagConfigurationName
-	ffConfig.Spec.FlagDSpec = &v1alpha1.FlagDSpec{Envs: []corev1.EnvVar{
-		{Name: "LOG_LEVEL", Value: "dev"},
-	}}
-	ffConfig.Spec.FeatureFlagSpec = featureFlagSpec
-	err = k8sClient.Create(testCtx, ffConfig)
-	Expect(err).ShouldNot(HaveOccurred())
-
-	fsConfig := &v1alpha1.FlagSourceConfiguration{}
-	fsConfig.Namespace = mutatePodNamespace
-	fsConfig.Name = flagSourceConfigurationName
-	fsConfig.Spec.Port = 8080
-	fsConfig.Spec.Evaluator = "yaml"
-	fsConfig.Spec.Image = "new-image"
-	fsConfig.Spec.Tag = "latest"
-	fsConfig.Spec.MetricsPort = 8081
-	fsConfig.Spec.SocketPath = "/tmp/flag-source.sock"
-	fsConfig.Spec.SyncProviderArgs = []string{
-		"key3=val3",
-	}
-	fsConfig.Spec.LogFormat = "console"
-	fsConfig.Spec.Sources = []v1alpha1.Source{
+func TestPodMutator_BackfillPermissions(t *testing.T) {
```

Comment:
@RealAnna mostly because I wanted some experience using the fake-client, I added a bit more coverage here that runs through the code to append subjects to the cluster role binding: https://github.com/RealAnna/open-feature-operator/pull/1

Let me know if these make sense, and if you want you can add them to this PR.